src/core_plugins/kibana/public/visualize/listing/visualize_listing_table.js

```diff
@@ -0,0 +1,267 @@
+  }
+
+  sortByTitle = () => this.sortOn('title');
+  sortByType = () => this.sortOn('type');
+
+  renderColumns() {
+    return [
+      <KuiTableHeaderCell
+        key="title"
+        onSort={this.sortByTitle}
+        isSorted={this.state.sortedColumn === 'title'}
+        isSortAscending={this.sortableProperties.isAscendingByName('title')}
+      >
+        Title
+      </KuiTableHeaderCell>,
+      <KuiTableHeaderCell
+        key="type"
+        onSort={this.sortByType}
+        isSorted={this.state.sortedColumn === 'type'}
+        isSortAscending={this.sortableProperties.isAscendingByName('type')}
+      >
+        Type
+      </KuiTableHeaderCell>
+    ];
+  }
+
+  renderRowCells(item) {
+    return [
+      <KuiTableRowCell key="title">
+        <a className="kuiLink" href={this.getUrlForItem(item)}>
+          {item.title}
+        </a>
+      </KuiTableRowCell>,
+      <KuiTableRowCell key="type">
+        <span className="kuiStatusText">
+          {this.renderItemTypeIcon(item)}
+          {item.type.title}
+        </span>
+      </KuiTableRowCell>
+    ];
+  }
+
+  createRows() {
+    return this.state.pageOfItems.map(item => ({
+      id: item.id,
+      cells: this.renderRowCells(item)
+    }));
+  }
+
+  closeModal = () => {
+    this.setState({ showDeleteModal: false });
+  };
+
+  renderConfirmDeleteModal() {
+    return (
+      <KuiModalOverlay>
+        <KuiConfirmModal
+          message="Are you sure you want to delete the selected visualizations? This action is irreversible!"
+          title="Warning"
+          onCancel={this.closeModal}
+          onConfirm={this.deleteSelectedItems}
+          cancelButtonText="Cancel"
+          confirmButtonText="Delete"
+        />
+      </KuiModalOverlay>
+    );
+  }
+
+  onDelete = () => {
+    this.setState({ showDeleteModal: true });
+  };
+
+  deleteSelectedItems = () => {
+    this.props.deleteSelectedItems(this.state.selectedRowIds)
+      .then(() => this.fetchItems(this.state.filter))
+      .catch(() => {})
+      .then(() => this.deselectAll())
+      .then(() => this.closeModal());
+  };
+
+  onItemSelectionChanged = (newSelectedIds) => {
+    this.setState({ selectedRowIds: newSelectedIds });
+  };
+
+  onCreate() {
+    window.location = '#/visualize/new';
+  }
+
+  renderToolBarActions() {
+    return this.state.selectedRowIds.length > 0 ?
+      <KuiListingTableDeleteButton onDelete={this.onDelete} aria-label="Delete selected visualizations"/> :
+      <KuiListingTableCreateButton onCreate={this.onCreate} aria-label="Create new visualization"/>;
+  }
+
+  renderPager() {
+    return (
+      <KuiPager
+        startNumber={this.state.pageStartNumber}
+        hasNextPage={this.pager.hasNextPage}
+        hasPreviousPage={this.pager.hasPreviousPage}
+        endNumber={this.pager.endItem}
+        totalItems={this.items.length}
+        onNextPage={this.onNextPage}
+        onPreviousPage={this.onPreviousPage}
+      />
+    );
+  }
+
+  renderPrompt() {
```

Comment:
BTW our style [recommends early exiting](https://github.com/elastic/kibana/blob/master/style_guides/js_style_guide.md#returnthrow-early-from-functions) so this would become:

```javascript
renderPrompt() {
  if (this.state.isFetchingItems) {
    return <KuiListingTableLoadingPrompt />;
  }

  if (this.items.length === 0) {
    if (this.state.filter) {
      return <KuiListingTableNoMatchesPrompt />;
    }
    
    return <NoVisualizationsPrompt />;
  }

  return null;
}
```

This makes it a little easier (for me at least) to mentally map conditions to outcome.